### DISCUSSION TITLE
TP 1 Ejercicio 5: Suma Lenta

### DISCUSSION THREAD

**Discussion by Elias-Turner** [INGCOM-UNRN-P1/cursada-2023#37] [Preguntas y respuestas] [answered]
Alguien tiene idea como hacer que la suma lenta haga 4+1+1+1+1.
Porque yo solo logre que hiciera: 
4 + 1= 5
4+2=6
4+3=7
4+4=8

**Comment by alengeor** [comment]
Como andas Elias?

Fijate, vos tenes dos numeros. Uno que se queda como está y otro que va a ser el que "descompongas" en uno's (1) y lo vayas sumando al primero.

Fijate que herramienta lógica podes usar para esto, pero tené en cuenta eso: hay uno de los dos numeros que vas a descomponer, e ir sumando de a uno _hasta_ que llegues a él. 😉

**Reply by martinvilu** [reply]
Un detalle, la salida en sí, es la suma. Lo que importa es como se hace la cuenta.

Por mas que la suma en sí sea `4+1+1+1`, el resultado va a seguir siendo `7`.

Esto porque los `printf` van en el `main`

**Reply by alengeor** [reply]
Podemos hacer printf en las funciones, a modo de "mostrar" que se va sumando de a 1? El resultado de la suma siempre será la salida de la función

**Reply by martinvilu** [reply]
Si, pero tiene que quedar en claro que el objetivo de la función no es el `printf`

**Comment by martinvilu** [comment]
> Alguien tiene idea como hacer que la suma lenta haga 4+1+1+1+1. Porque yo solo logre que hiciera: 4 + 1= 5 4+2=6 4+3=7 4+4=8

Esto es correcto, siempre y cuando, internamente, la suma se haga `4+1+1+1+1`

**Comment by martinvilu** [comment, marked as answer]
Ah, no olviden marcar la pregunta como contestada, y que la suma lenta es para números enteros, que incluyen números negativos en ambos términos.
